libs/Common/CMakeLists.txt

```diff
@@ -13,6 +13,7 @@ endif()
 
 # Link its dependencies
 TARGET_LINK_LIBRARIES(Common ${Boost_LIBRARIES} ${OpenCV_LIBS})
+TARGET_COMPILE_OPTIONS(Common INTERFACE -fpermissive)
```

Comment:
pls detail, I did not understand the comment from the PR; at line 3062 all looks fine to me:
```
			fImage.write(cv::Mat::template ptr<const float>(--i), rowbytes);
```

Reply:
Unfortunatelly either new version of OpenCV has some kind of a bug or GCC12 is behaving differently and selecting non const template version of a ptr function and though const qualifier is not respected for this. I can paste exact error message if you are able to help with that. I tried to solve this myself, but I am not so smart, so I removed this code, as I am not using it anywhere (PFM image files). As I understand `-fpermissive` will remove such restrictions.

 ```
[ 24%] Building CXX object libs/IO/CMakeFiles/IO.dir/OBJ.cpp.o
cc1plus: warning: command-line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
In file included from /home/jbialas/openMVS/libs/Common/Types.h:2789,
				 from /home/jbialas/openMVS/libs/Common/Common.h:176,
				 from /home/jbialas/openMVS/libs/IO/Common.h:18,
				 from /home/jbialas/openMVS_build/libs/IO/CMakeFiles/IO.dir/cmake_pch.hxx:5,
				 from <command-line>:
/home/jbialas/openMVS/libs/Common/Types.inl: In instantiation of ‘bool SEACAVE::TImage<TYPE>::Save(const SEACAVE::String&) const [with TYPE = SEACAVE::TPixel<unsigned char>]’:
/home/jbialas/openMVS/libs/IO/OBJ.cpp:77:39:   required from here
/home/jbialas/openMVS/libs/Common/Types.inl:3062:72: error: passing ‘const SEACAVE::TImage<SEACAVE::TPixel<unsigned char> >’ as ‘this’ argument discards qualifiers [-fpermissive]
 3062 |						 fImage.write(cv::Mat::template ptr<const float>(--i), rowbytes);
	  |									  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
In file included from /usr/include/opencv4/opencv2/core/mat.hpp:3773,
				 from /usr/include/opencv4/opencv2/core.hpp:58,
				 from /usr/include/opencv4/opencv2/opencv.hpp:52,
				 from /home/jbialas/openMVS/libs/Common/Types.h:143:
/usr/include/opencv4/opencv2/core/mat.inl.hpp:713:6: note:   in call to ‘_Tp* cv::Mat::ptr(int) [with _Tp = const float]’
  713 | _Tp* Mat::ptr(int y)
	  |	  ^~~
make[2]: *** [libs/IO/CMakeFiles/IO.dir/build.make:237: libs/IO/CMakeFiles/IO.dir/OBJ.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:405: libs/IO/CMakeFiles/IO.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
  
```

Reply:
@cdcseacave I'm having the same issue on Fedora 36 and `gcc 12.1.1 20220507 (Red Hat 12.1.1-1)` when building the `v2.0.1` tag, with `boost 1.76` and and latest `vcglib`.

```
OPENMVS_VERSION="2.0.1"

# Configure CMake
cmake "../third_party/openMVS-${OPENMVS_VERSION}" \
  -DCMAKE_BUILD_TYPE=Release \
  -DBUILD_SHARED_LIBS=OFF \
  -DCMAKE_INSTALL_PREFIX=/opt/3dr \
  -DBOOST_ROOT="../third_party/boost_1_76_0/stage" \
  -DVCG_ROOT="../third_party/vcglib" \
  -DBoost_USE_STATIC_LIBS=ON
```